examples/jsm/nodes/utils/PackingNode.js

```diff
@@ -0,0 +1,40 @@
+import TempNode from '../core/TempNode.js';
+import { normalize } from '../shadernode/ShaderNodeBaseElements.js';
+
+class PackingNode extends TempNode {
+
+	constructor( scope, node ) {
+
+		super( 'vec3' );
+
+		this.scope = scope;
+		this.node = node;
+
+	}
+
+	construct() {
+
+		const { scope, node } = this;
+
+		let result = null;
+
+		if ( scope === PackingNode.NORMAL_TO_RGB ) {
+
+			result = normalize( node ).mul( 0.5 ).add( 0.5 );
+
+		} else if ( scope === PackingNode.RGB_TO_NORMAL ) {
+
+			result = node.mul( 2.0 ).sub( 1 );
+
+		}
+
+		return result;
+
+	}
+
+}
+
+PackingNode.NORMAL_TO_RGB = 'normalToRGB';
```

Comment:
This node just performs [-1, 1] <-> [0, 1] conversion which is useful in many contexts -- not sure that we should limit it to only normals and colors... Maybe something like `mapPositive`/`unmapPositive`?

Reply:
Sounds good. Maybe `pack/unpack` instead of `map/unmap`?

Reply:
It's more a mapping than a packing, I think...

Reply:
Or `directionToRGB`.. we shouldn’t use `mapPosition` to colors.

Reply:
Maybe `mapToUnitInterval` and `unmapFromUnitInterval`? Slightly more verbose but more understandable.

Reply:
I'm in doubt if we use unit length for something besides direction? Sorry to insist so much, just to exhaust the possibilities.

Reply:
I mean not unit length but the unit interval ([0, 1]).

Reply:
I'm referring to the original value [ -1, 1 ]. The conversion will only be correct if the input vector is in this format.

Reply:
I think you use `normalize` there? So it would be converted into that format.

Reply:
> I think you use normalize there? So it would be converted into that format.

normalize() is redundant, this can be removed, we will not have other compatible values if not unity length, normalize() will flatten values above 1. `directionToColor` seems more appropriate to me in this logic.

Reply:
OK, I'm fine with `directionToColor`.

Reply:
Great, I'll update, it was a good catch.